src/entities/entity.cpp

```diff
@@ -0,0 +1,114 @@
+/***
+ * Maracatronics Robotics
+ * Federal University of Pernambuco (UFPE) at Recife
+ * http://www.maracatronics.com/
+ *
+ * This file is part of Armorial project.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ***/
+
+#include "entity.h"
+
+Entity::Entity(EntityType type) {
+    _entityType = type;
+    _entityPriority = 0; // default priority is 0
+    _loopFrequency = 60; // default loop frequency is 60
+    _isEnabled = true;   // enabling by default
+}
+
+void Entity::run(){
```

Comment:
This is an inherited method from the "QThread" class. When creating an Entity from this class, just call the **start()** (also from the QThread) method and it will run this method in a loop.